src/sdk/namespace_fs.js

```diff
@@ -496,15 +495,22 @@ class NamespaceFS {
     }
 
     async _upload_stream(source_stream, upload_path, write_options) {
-        return new Promise((resolve, reject) =>
-            source_stream
-            .once('error', reject)
-            .pipe(
-                fs.createWriteStream(upload_path, write_options)
-                .once('error', reject)
-                .once('finish', resolve)
-            )
-        );
+        let target_file;
+        try {
+            target_file = await nb_native().fs.open(DEFAULT_FS_CONFIG, upload_path, 'a');
+            for await (const data of source_stream) {
+                await target_file.write(DEFAULT_FS_CONFIG, data);
+            }
```

Comment:
We need to think a bit on how to optimize this write flow.
The source_stream is typically the HTTP request (although it might be a copy object stream too).
This means that the data chunks we will get from that stream might be quite small.
You can print data.length and see for yourself.
In the case of small buffers we would want to collect up to `config.NSFS_BUF_SIZE` bytes and submit a single writev call.
https://linux.die.net/man/2/writev

Reply:
Should it be done in this PR?